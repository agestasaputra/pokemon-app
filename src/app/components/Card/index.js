import React from 'react'
import { NavLink } from "react-router-dom"

export default function Card({ pokemon, onCardClicked }) {
  return (
    <>
      {
        pokemon.list.map((data, key) => (
          <NavLink to={`/${key+1}`} onClick={onCardClicked} key={key}>
            <div className="card" >
              <div className="card-info">
                <h6 className="card-info__title">
                  { data.name }
                </h6>
                <p className="card-desc">
                  Owned: <strong>{ data.owned }</strong>
                </p>
              </div>
              <img src={data.image} alt={data.name} />
            </div>
          </NavLink>
        ))
      }
    </>
  )
}
