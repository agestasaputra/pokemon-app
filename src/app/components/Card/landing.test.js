import React from 'react';
import { create } from 'react-test-renderer'
import Card from './index'
import { BrowserRouter } from "react-router-dom";

test("renders when there is one item", () => {
  const pokemon = {
    list: [
      {
        customName: "",
        id: 1,
        image: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png",
        name: "bulbasaur",
        owned: 0,
        url: "https://pokeapi.co/api/v2/pokemon/1/"
      }
    ],
    loading: false,
    next: "https://pokeapi.co/api/v2/pokemon?offset=20&limit=20"
  }
  const tree = create(
    <BrowserRouter>
      <Card pokemon={pokemon} />
    </BrowserRouter>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})