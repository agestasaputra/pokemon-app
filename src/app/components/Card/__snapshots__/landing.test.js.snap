// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there is one item 1`] = `
<a
  aria-current={null}
  href="/1"
  onClick={[Function]}
>
  <div
    className="card"
  >
    <div
      className="card-info"
    >
      <h6
        className="card-info__title"
      >
        bulbasaur
      </h6>
      <p
        className="card-desc"
      >
        Owned: 
        <strong>
          0
        </strong>
      </p>
    </div>
    <img
      alt="bulbasaur"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
    />
  </div>
</a>
`;
